File: output/ChatGPT/avatar/Python/C++/codeforces_621_A.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> lst(n);
    for (int i = 0; i < n; i++) {
        cin >> lst[i];
    }
    sort(lst.begin(), lst.end(), greater<int>());
    int od = 0;
    for (int i = 0; i < n; i++) {
        if (lst[i] % 2 != 0) {
            od++;
        }
    }
    int sum = 0;
    bool ok = false;
    for (int i = 0; i < n; i++) {
        if (lst[i] % 2 == 0) {
            sum += lst[i];
        } else {
            if (ok || od > 1) {
                sum += lst[i];
                od--;
                ok = !ok;
            }
        }
    }
    cout << sum << endl;
    
    return 0;
}

File: output/CodeGen/avatar/Python/Java_handraft/codeforces_621_A.java
import java.util.*;

public class codeforces_621_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt (scanner.nextLine());
        String[] input = scanner.nextLine().split(" ");
        System.out.println(findMaxForm(input, n));
    }
    public static long findMaxForm(String[] strs, int n) {
        Long[] lst = new Long[n];
        for (int i = 0; i < n; i++) {
            lst[i] = Long.parseLong(strs[i]);
        }
        Arrays.sort(lst);
        Arrays.sort(lst, Collections.reverseOrder());
        long sum = 0;
        boolean ok = false;
        int od = 0;
        for (long i : lst) {
            if (i % 2 != 0) {
                od += 1;
            }
        }
        for (int i = 0; i < n; i++) {
            if (lst[i] % 2 == 0) {
                sum += lst[i];
            } else {
                if (ok || od > 1) {
                    sum += lst[i];
                    od--;
                    ok =!ok;
                }
            }
        }
        return sum;
    }
}